examples/bri-3/src/bri/communication/messagingClients/natsMessagingClient.ts

```diff
@@ -0,0 +1,61 @@
+import { Injectable, OnModuleDestroy, OnModuleInit } from '@nestjs/common';
+import { connect, NatsConnection, StringCodec } from 'nats';
+import { LoggingService } from '../../../shared/logging/logging.service';
+import { IMessagingClient } from './messagingClient.interface';
+
+@Injectable()
+export class NatsMessagingClient
+  implements IMessagingClient, OnModuleInit, OnModuleDestroy
+{
+  private nc: NatsConnection;
+
+  constructor(private readonly log: LoggingService) {}
+
+  async onModuleInit() {
+    try {
+      this.nc = await connect({ servers: process.env.BPI_NATS_SERVER_URL });
+      this.log.logInfo('Connected to nats server: ' + this.nc.getServer());
+    } catch (err) {
+      this.log.logWarn(
+        `Error: ${err} connecting to nats server at: ${process.env.BPI_NATS_SERVER_URL}`,
+      );
+    }
+  }
+
+  async onModuleDestroy() {
+    if (this.nc && !this.nc.isClosed()) {
+      await this.nc.drain();
+    }
+  }
+
+  async subscribe(
+    channelName: string,
+    callback: (message: string) => void,
+  ): Promise<void> {
+    if (!this.nc || this.nc.isClosed()) {
+      this.log.logWarn(`Trying to subscribe without an open nats connection`);
+      return;
+    }
+
+    const sc = StringCodec();
```

Comment:
same as nc, but if you want to keep it like this its ok for me, but please rename `const subscription` to `const sbc` in that case :D

Reply:
oh sweet baby jesus

Reply:
Fixed